### ISSUE TITLE
gossip: Random test failure in gossip test "test filtering values works"

### ISSUE THREAD

**Issue by InKryption** [Syndica/sig#157]
### Description

At random, this line in the referenced test will fail:

https://github.com/Syndica/sig/blob/a5d1f34d8a93f0d101e58e2a5aa2a8a5126a632f/src/gossip/pull_response.zig#L143

I currently do not have a stacktrace demonstrating this behavior, because I do not know the circumstances under which this failure occurs.

### How to Reproduce the Bug

run `zig build test -Dfilter="gossip.pull_response: test filtering values works"` and pray

### Additional Context

It should be noted that the function being tested, named `filterSignedGossipDatas`, currently instantiates and uses a PRNG seeded with `std.time.milliTimeStamp()`, which could explain its seemingly random and unreproducible behavior.

What would be ideal is capturing a stack trace of this occurring, as well as the exact or approximate timestamp at which this failure occurs, to diagnose what random values can cause such a failure, and identify what circumstances they lead to which must be guarded against.

At the time of writing I plan on pushing a change which would help in the above goals, without needing to actively search for this mythical seed.

**Comment by InKryption** [comment]
Sorry, I'd like to keep this open until we actually have it solved. The code for helping us solve it is in place, it just needs to randomly occur at some point, or the code needs to fundamentally change such that we can guarantee it won't.

**Comment by 0xNineteen** [comment]
just hit it - The failing seed is: '1718031575764'

**Comment by InKryption** [comment]
Can confirm that using '1718031575764' as the seed consistently reproduces the failure state.

**Comment by dnut** [comment]
another one: 1718158590375

**Comment by InKryption** [comment]
Can't reproduce the behavior using '1718158590375'.

**Comment by 0xNineteen** [comment]
yeah pulling out the random seeds to deterministic would be good - we can create a ticket - kind of low priority right now

**Comment by InKryption** [comment]
New failing seed caught in CI:
https://github.com/Syndica/sig/actions/runs/9811929838/job/27094994026?pr=189
Seed number is '1720202203195'.

**Comment by InKryption** [comment]
https://github.com/Syndica/sig/actions/runs/9897406056/job/27341819918#step:4:1

We got another one: '1720726415143'

**Comment by dnut** [comment]
This is still a problem
